### ISSUE TITLE
Discussion: Inline code block disabled

### ISSUE THREAD

**Issue by DarinDev1000** [RocketChat/Rocket.Chat#26865]
Currently, to make a code block in a message, the 3 backticks on each end have to be on a new line.
I don't mind it being that way but when importing messages from slack, slack supports inline code blocks and they do not convert correctly.
Is this an intended feature?
Maybe inline full code blocks should be supported?

Example:
````
This works in chat:

```
{
    "code": "block"
}
```

  
This does not work:

```{
    "code": "block"
}```
````

**Comment by dudanogueira** [comment]
Hi! Thanks for pointing it out.

I marking here for discussion. I believe that the way it's working at with new parser follows a more strict approach, considering that right after the initial 3 back ticks, a convention is to define the name of the programming language to be rendered above., like so 

replacing ` for ':

''' python
def populate_config(self):
    pass
'''

**Comment by hugocostadev** [comment]
I agree with @dudanogueira , I tried to write the same code that you send in the issue here and it does not reproduce the mentioned output:

This can generate a lot of issues, in my opinion, we should follow the current implementation. 

**Input:**
```
```{
    "code": "block"
}```
```

**Outputs:**
```{
    "code": "block"
}```

**Comment by sketch242** [comment]
In addition to user expectations for people who are coming from slack or have been using rocket.chat for years, this also breaks message history for anyone using the new message template.  We have years of messages that are still visible in search, but no longer format properly.  

Some users also have the legacy message template enabled and can still send messages like this, but they don't format correctly for people on the new message template.

This was marked as fixed in #26388, but it's not as of 5.3.1.

**Comment by ghuls** [comment]
This is how I fixed the code block from a Slack export before importing it in RocketChat:

```bash
mkdir slack_export
cd slack_export

# Unzip your original slack export zip archive.
unzip ../original_slack_export.zip

# Fix "```" codeblock (that lack new lines before and/or after the "```" in the json files of Slack export.
sed -i -e '/^ \+"text":/ {s/\(\\n\)\?```\(\\n\)\?/\\n```\\n/g; s/```\\n",$/```",/; }' */*.json

# Zip fixed slack export JSON files
zip ../fixed_codeblocks_slack_export.zip -r .
```

Also the Mongo DB needs to be fixed manually for now to actually see any messages in the imported channels:
https://github.com/rocketchat/rocket.chat/issues/37058

**Comment by ghuls** [comment]
The typescript equivalent of the previous `sed` code should likely be added after this line:

https://github.com/RocketChat/Rocket.Chat/blob/develop/apps/meteor/app/importer-slack/server/SlackImporter.ts#L375

**Comment by ghuls** [comment]
This will likely fix it, on import:

```typescript
diff --git a/apps/meteor/app/importer-slack/server/SlackImporter.ts b/apps/meteor/app/importer-slack/server/SlackImporter.ts
index 87098c8b35..788d1e8553 100644
--- a/apps/meteor/app/importer-slack/server/SlackImporter.ts
+++ b/apps/meteor/app/importer-slack/server/SlackImporter.ts
@@ -692,6 +692,9 @@ export class SlackImporter extends Importer {
                        message = message.replace(/<#([^|]*)\|([^>]*)>/g, '#$2');
                        message = message.replace(/<@([^|]*)\|([^>]*)>/g, '@$1');
                        message = message.replace(/<@([^|>]*)>/g, '@$1');
+                       message = message.replace(/(\n)?```(\n)?/g, '\n```\n');
+                       message = message.replace(/^\n```/, '```');
+                       message = message.replace(/```\n$/, '```');
                } else {
                        message = '';
                }
```

**Comment by ghuls** [comment]
### Description:



### Steps to reproduce:

1. Import Slack archive: users and channels
2. Fix MongoDB so imported messages show up in channels (otherwise channels look empty). Multiple options:
   - manually run commands in mongosh: https://github.com/RocketChat/Rocket.Chat/issues/37058#issuecomment-3342623466
   - wait for #37805 to be update with this fix (or a better one): https://github.com/RocketChat/Rocket.Chat/pull/37805#issuecomment-3743637660
   - Get the database migraton from https://github.com/RocketChat/Rocket.Chat/pull/38090 running after the import to fix MongoDb database (same commands ran as in manual mongosh step.
3. Go to a channel with code block where in Slack "```" code blocks are not separated by newline before and after.



### Expected behavior:

I expect that code blocks from Slack imports are rendered properly even if they are not how RocketChat expects them (separated by newlines). Slack does not support specifying the language for a code block (e.g. "```python") so there is no ambiguity like in RocketChat.

### Actual behavior:

All code blocks that don't have "```" separated by a newline before or after are not rendered as a code block.

### Server Setup Information:

- Version of Rocket.Chat Server: 7.13.2
- Operating System: Ubuntu
- Deployment Method: Docker
- Number of Running Instances: 1
- DB Replicaset Oplog: iredTiger (oplog Enabled)
- NodeJS Version: v22.16.0
- MongoDB Version: 6.0.13

### Client Setup Information

- Desktop App or Browser Version: In all browsers
- Operating System: On all operating systems

**Comment by ghuls** [comment]
Pull request #38160 fixes this Slack inline code block on import to RocketChat approved code blocks.
